zebra-chain/src/commitments/sapling.rs

```diff
@@ -0,0 +1,291 @@
+//! Sapling note and value commitments and types.
+
+#![allow(clippy::unit_arg)]
+
+#[cfg(test)]
+mod arbitrary;
+
+use std::{fmt, io};
+
+use bitvec::prelude::*;
+use rand_core::{CryptoRng, RngCore};
+
+use crate::{
+    keys::sapling::{find_group_hash, Diversifier, TransmissionKey},
+    serde_helpers,
+    serialization::{ReadZcashExt, SerializationError, ZcashDeserialize, ZcashSerialize},
+    types::amount::{Amount, NonNegative},
+};
+
+/// Generates a random scalar from the scalar field 𝔽_{r_𝕁}.
+///
+/// The prime order subgroup 𝕁^(r) is the order-r_𝕁 subgroup of 𝕁 after the
+/// Edwards cofactor h_𝕁 = 8 is factored out. This function is useful when
```

Comment:
nit: I think the "factored out" part of this sentence is strictly speaking not quite right, since it implies that the prime-order subgroup is a quotient of the full group by the small-order subgroup. But although there's a correspondence between the two, they're not exactly the same (one has points and the other has equivalence classes of points).

Instead I think it would be better to say just that the prime-order subgroup consists of the points whose order divides r.